src/RtcTemperature.h

```diff
@@ -7,32 +7,54 @@
 class RtcTemperature
 {
 public:
-    RtcTemperature(int8_t degrees, uint8_t fraction) :
-        integerDegrees(degrees),
-        decimalFraction(fraction)
+    RtcTemperature( int8_t msdegC, uint8_t lsdegC ) :
+        mstemp( msdegC ),
+        lstemp( lsdegC & 0xC0 )  // Bind to RTC resolution
     {
     }
 
+    // Scaled integer temperature (x 256)
+    int16_t AsScaledDegrees()
+    {
+        return(  (mstemp << 8) + lstemp  );  // Concatentate raw temp. regs 
+    }
+
+    // Rounded integer temperature (x 1)
+    int8_t AsRoundedDegrees()
+    {
+        // Equivalent float function:
+        //   SF ( sign(x)    * floor(  abs(       x ) + 0.5    ) ) / SF, with
+        // =    ( sign(SF x) * floor(  abs(    SF x ) + 0.5 SF ) ) / SF  SF = 256
+        // =      sign(x)    * floor( sign(x)( SF x ) + 0.5 SF )   / SF
+        //
+        int8_t  sgnT = ( (mstemp < 0) ? -1 : 1 );
+        return( sgnT * ((sgnT * AsScaledDegrees() + 0x80) >> 8) );
+    }
+
+    // Float temperature (x 1)
     float AsFloat()
     {
-        float degrees = (float)integerDegrees;
-        degrees += (float)decimalFraction / ((degrees < 0) ? -100.0f : 100.0f) ;
-        return degrees;
+        return ( (float)AsScaledDegrees() / 256.0f );
     }
 
+    // Integer portion, temperature (x 1), useful for printing
     int8_t AsWholeDegrees()
     {
-        return integerDegrees;
+        int8_t  sgnT = ( (mstemp < 0) ? -1 : 1 );
+        return( sgnT * ((sgnT * AsScaledDegrees()) >> 8) ); // (x 1)
```

Comment:
Why do the extra work within AsScaledDegrees to just undo it here.  Just return what you are currently calling mstemp.